cpp/src/io/parquet/experimental/deletion_vectors.cu

```diff
@@ -156,13 +258,152 @@ std::unique_ptr<cudf::column> build_row_mask_column(
 
 }  // namespace
 
-table_with_metadata read_parquet_and_apply_deletion_vector(
+/**
+ * @copydoc
+ * cudf::io::parquet::experimental::chunked_parquet_reader::chunked_parquet_reader
+ */
+chunked_parquet_reader::chunked_parquet_reader(
+  std::size_t chunk_read_limit,
+  std::size_t pass_read_limit,
+  parquet_reader_options const& options,
+  cudf::host_span<cuda::std::byte const> serialized_roaring64,
+  cudf::host_span<size_t const> row_group_offsets,
+  cudf::host_span<size_type const> row_group_num_rows,
+  rmm::cuda_stream_view stream,
+  rmm::device_async_resource_ref mr)
+  : _start_row{0},
+    _is_unspecified_row_group_data{row_group_offsets.empty()},
+    _stream{stream},
+    _mr{mr},
+    // Use default mr for the internal chunked reader and row index column if we will
+    // be applying the deletion vector to produce the output table
+    _table_mr{serialized_roaring64.empty() ? mr : rmm::mr::get_current_device_resource_ref()}
```

Comment:
Regarding `table_mr`, if it’s not an output, similar use cases like `hash_join` typically just uses the current device mr without requiring it to be passed as an argument.